### DISCUSSION TITLE
question: start_http_server - how to prevent it from accepting external connection?

### DISCUSSION THREAD

**Discussion by yxiang92128** [prometheus/client_python#808] [Q&A]
As part of our InfoSec hardening requirement, we need to make sure this REST server does NOT accept any connection from an external IP other than internally from within the same host. Is there a parameter we can pass in or anything we could to do to make sure of this?

thanks

**Comment by csmarchbanks** [comment]
Hello, I converted this to a discussion as it is a question not an issue with the library.

To make sure the server created by `start_http_server` cannot be reached by external services you can specify the `addr` to be used. For example,
```python
start_http_server(8000, addr="127.0.0.1")
```
would start a server on port 8000 that can only be accessed locally.
